torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -270,17 +486,18 @@ class IrParser {
                           BinaryOpType::Sub,
                           static_cast<BinaryOpWithAlphaType>(&sub_alpha))}});
             // TODO: handle scaling factor when it's not constant 1;
-            auto lhs = value_map[node->inputs()[0]->unique()];
-            auto rhs = value_map[node->inputs()[1]->unique()];
-            auto alpha = value_map[node->inputs()[2]->unique()];
 
-            if (alpha->isOneInt()) {
-              auto out = binaryOp(op_mapping[node->kind()].first, lhs, rhs);
-              value_map.emplace(node->output()->unique(), out);
-            } else {
-              auto out = op_mapping[node->kind()].second(lhs, rhs, alpha);
-              value_map.emplace(node->output()->unique(), out);
-            }
+            auto [format, lhs, rhs] = getConsistentValues(
```

Comment:
Just wanted to quickly confirm if I'm understanding correctly. In these multi-input ops if `c10::nullopt` is passed as format, the output will be c_last if any one of the input is c_last, would that be correct?

Reply:
Also maybe I missed this part but would we be able to handle implicit broadcast on channel dimension in this case ? e.g. 
```
 A[N,C,H,W] + B[H,W]
```

Reply:
Good catch!!! I don't think it works, I need to update python test to cover that.

As a quick WAR, we could be disabling format propagation for implicit broadcast and permute all inputs to contiguous.
Alternatively, we can unsqueeze contiguous tensors, so they would have a broadcast dim towards the end... I need to go over the list of ops to double check if there's any exceptions.